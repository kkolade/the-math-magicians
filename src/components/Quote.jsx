import React, { useEffect, useState } from 'react';

const Quote = () => {
  const [quote, setQuote] = useState([]);
  const [hasError, setHasError] = useState(false);
  const [isLoading, setIsLoading] = useState(false);
  const url = 'https://api.api-ninjas.com/vi/quotes';

  useEffect(() => {
    const getQuote = async () => {
      setIsLoading(true);
      try {
        const fetchQuote = await fetch(url, {
          method: 'GET',
          header: {
            'Content-Type': 'application/json',
            'X-Api-Key': 'tgGQLF0ayPX9ZzYdnazDEA==ZUlyDiE4HJM8YHuy',
          },
        });
        const fetchedQuote = fetchQuote.json();
        console.log(fetchedQuote);
        setQuote(fetchedQuote);
      } catch (error) {
        setHasError(true);
        console.log(error);
      }
      setIsLoading(false);
    };
    getQuote();
  }, [setQuote, setIsLoading]);

  if (hasError) {
    return (
      <div className="Quotes-container">
        <span className="error">Error getting the quotes!!!</span>
      </div>
    );
  }
  if (isLoading) {
    return (
      <div className="Quotes-container">
        <span>Loading the quotes...</span>
      </div>
    );
  }
  return (
    <div className="Quotes-container">
      <p>
        &quot;
        {quote[0].quote}
        &quot;
      </p>
      <p>{quote[0].author}</p>
    </div>
  );
};

export default Quote;
