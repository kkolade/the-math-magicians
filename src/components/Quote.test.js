import React from 'react';
import renderer from 'react-test-renderer';
import Quote from './Quote';

describe('Quote component', () => {
  it('should render correctly', () => {
    const component = renderer.create(<Quote />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
