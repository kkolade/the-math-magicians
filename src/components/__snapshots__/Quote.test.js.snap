// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote component should render correctly 1`] = `
<div
  className="Quote-container"
>
  <p>
    "
    "
  </p>
  <p
    className="Quote-author"
  />
</div>
`;

exports[`renders an error message if the API call fails 1`] = `
<span
  class="error"
>
  Error getting quote!!!
</span>
`;
